### PR TITLE
[docs] Change default date from 4th of April to 17th of April for readability

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

**Review by LukasTy — CHANGES_REQUESTED** [review]
This is a really great readability improvement! ❤️ 
As someone, who "can only read" ISO dates I appreciate it very much. 👏 
Superb docs feedback point.

File: docs/data/date-pickers/base-concepts/ComponentFamilies.tsx [hunk 1]
```diff
@@ -39,36 +39,36 @@ export default function ComponentFamilies() {
...
-          <DateTimeField defaultValue={dayjs('2022-04-07T15:30')} />
+          <DateTimeField defaultValue={dayjs('2022-04-21T15:30')} />
         </DemoItem>
         <DemoItem
           label={<ProLabel>Date Range</ProLabel>}
           component="MultiInputDateRangeField"
         >
           <MultiInputDateRangeField
-            defaultValue={[dayjs('2022-04-07'), dayjs('2022-04-10')]}
+            defaultValue={[dayjs('2022-04-21'), dayjs('2022-04-10')]}
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/date-pickers/base-concepts/ComponentFamilies.tsx:55]: :facepalm: ok I was too fast on this one...

[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/date-pickers/base-concepts/ComponentFamilies.tsx:55]: I'll set `04-17` as the default date and `04-21` as the 2nd date on range components

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ describe('<AdapterDayjs />', () => {
 
     it('should not create UTC dates when no instance passed', () => {
       const onChange = spy();
-      render(<DateCalendar defaultValue={dayjs('2022-04-10')} onChange={onChange} />);
+      render(<DateCalendar defaultValue={dayjs('2022-04-21')} onChange={onChange} />);
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx:86]: If we are aligning everywhere, then maybe going with the following would be a bit more correct? 🤔 
```suggestion
      render(<DateCalendar defaultValue={dayjs('2022-04-17')} onChange={onChange} />);
```

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx [hunk 3]
```diff
@@ -96,7 +96,7 @@ describe('<AdapterDayjs />', () => {
       const onChange = spy();
       render(
         <LocalizationProvider dateAdapter={AdapterDayjs} dateLibInstance={dayjs.utc}>
-          <DateCalendar defaultValue={dayjs.utc('2022-04-10')} onChange={onChange} />
+          <DateCalendar defaultValue={dayjs.utc('2022-04-21')} onChange={onChange} />
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx:99]:
```suggestion
          <DateCalendar defaultValue={dayjs.utc('2022-04-17')} onChange={onChange} />
```

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx [hunk 4]
```diff
@@ -116,7 +116,7 @@ describe('<AdapterDayjs />', () => {
       const { executeDateDrag } = buildPickerDragInteractions(() => dataTransfer);
 
       const onChange = spy();
-      const initialValue: [any, any] = [dayjs.utc('2022-04-10'), dayjs.utc('2022-04-12')];
+      const initialValue: [any, any] = [dayjs.utc('2022-04-21'), dayjs.utc('2022-04-21')];
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx:119]:
```suggestion
      const initialValue: [any, any] = [dayjs.utc('2022-04-17'), dayjs.utc('2022-04-21')];
```

**Review by LukasTy — APPROVED** [review]
Great work! 👏 
Very nice result and a few optional alignment improvements so that all the date and times are the same everywhere. 👍

**Pull request merged**